saas/api/roles.py

```diff
@@ -944,3 +946,40 @@ def destroy(self, request, *args, **kwargs):
                 'organization': self.organization.slug, 'roles': roles})
         queryset.delete()
         return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+class RoleAcceptAPIView(CreateAPIView):
+
+    serializer_class = AcceptRoleSerializer
+
+    def perform_create(self, serializer):
+        request = self.request
+        obj = get_object_or_404(get_role_model().objects.all(),
+                grant_key=serializer.validated_data['verification_key'])
+        existing_role = get_role_model().objects.filter(
+            organization=obj.organization, user=request.user).exclude(
+            pk=obj.pk).first()
+        if existing_role:
+            raise ValidationError(_("You already have a %(existing_role)s"\
+                " role on %(organization)s. Please drop this role first if"\
+                " you want to accept a role of %(role)s instead.") % {
+                    'role': obj.role_description.title,
+                    'organization': obj.organization.printable_name,
+                    'existing_role': existing_role.role_description.title})
+
+        obj.user = request.user       # We appropriate the Role here.
```

Comment:
This cannot be happening here. That code is specific to a user clicking on a link in the invite e-mail and then login using a different account than the one invited.